packages/components/src/DataTable/Item/DataTableRow.tsx

```diff
@@ -87,6 +87,9 @@ const DataTableRowLayout = forwardRef(
         ? undefined
         : onClick && onClick(event)
 
+    const handleSecondaryClick = (event: React.MouseEvent<HTMLElement>) => {
+      event.stopPropagation()
+    }
```

Comment:
I think it'd be useful to clearly name this for it's use-case (suppressing the click to that parent record)

```suggestion
    const suppressClickPropagation = (event: React.MouseEvent<HTMLElement>) => {
      event.stopPropagation()
    }
```